### DISCUSSION TITLE
is select a memory leak?

### DISCUSSION THREAD

**Discussion by systemlord01** [ngxs/store#1722] [Q&A] [answered]
Just a quick question. Since you do not unsubscribe in your select decorator, I think it might be a memory leak.

I could be wrong. Didn't go to deep into the code. But I am curious how it is unsubscribed.

**Comment by markwhitfeld** [comment, marked as answer]
The `@Select` decorator merely makes the observable available at that property. It does not subscribe.
If you subscribe then you would have to manage that subscription, but I would recommend using the `async` pipe to manage your subscription in the template.

**Reply by systemlord01** [reply]
Ah yes very true. Went to deep into the implementation without looking at usage. My bad. Thx for the input.
